app/models/job.rb

```diff
@@ -14,6 +14,14 @@ class Job < ActiveRecord::Base
   default_scope { by_newest }
 
   def self.visible_for(user)
-    all.where.not(id: user.jobs_hidden.pluck(:id))
+    _not_hidden(user)._not_liked(user)
```

Comment:
Sorry for the intrusion back-end team, we changed the behavior of the .visible_for method so that it only returns jobs that aren't hidden and aren't liked for that user. That way when you like a job it removes it from "All Jobs" and you can "clean" out your inbox in a way. Otherwise the "All Jobs" would grow and could never be cleared".

Seems redundant to have it added to "My Jobs" and then keep it in "All Jobs".

Also broke out the class methods into two specific filters.